src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -774,28 +776,44 @@ internal async Task SearchPackagesAndRefreshUpdateCountAsync(string searchText,
                 }
                 else
                 {
-                    RefreshAvailableUpdatesCount();
+                    RefreshInstalledAndUpdatesTabs();
                 }
             }
         }
 
-        private void RefreshAvailableUpdatesCount()
+        private void RefreshInstalledAndUpdatesTabs()
         {
             NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
             {
                 await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
+                _topPanel._labelInstalled.ShowWarning = false;
                 _topPanel._labelUpgradeAvailable.Count = 0;
                 var loadContext = new PackageLoadContext(ActiveSources, Model.IsSolution, Model.Context);
                 var packageFeed = await CreatePackageFeedAsync(loadContext, ItemFilter.UpdatesAvailable, _uiLogger);
                 var loader = new PackageItemLoader(
                     loadContext, packageFeed, includePrerelease: IncludePrerelease);
+                var metadataProvider = CreatePackageMetadataProvider(loadContext);
 
-                // cancel previous refresh update count task, if any
+                // cancel previous refresh tabs task, if any
                 // and start a new one.
                 var refreshCts = new CancellationTokenSource();
                 Interlocked.Exchange(ref _refreshCts, refreshCts)?.Cancel();
 
+                // Update installed tab warning icon
+                var installedDeprecatedPackagesCount = await GetInstalledDeprecatedPackagesCountAsync(
```

Comment:
Is this cached like the rest of the models?

Reply:
The information that the UI needs to display the tab is cached in `_topPanel._labelInstalled`. I didn't think caching any additional information about the number of installed deprecated packages would provide any additional benefits.

Reply:
Also, it looks like `loadContext.GetInstalledPackagesAsync` caches the result of fetching installed packages internally, so most of the time this will be a very cheap operation.